### PR TITLE
Encrypted dynamo

### PR DESCRIPTION
**Pull request by pvighi**
added a cloudformation for the new encrypted dynamo tables and change the code and original cloudformation to refer to the new ones.
Tested in CODE and it seems to work.
As part of this test the CODE and DEV tables have been created. 

**Before merging**
Before merging this PR we should:

- Create the encrypted prod table running the tables cloudformation

- Copy all data from the non encrypted table into it
- Run the main Cloudformation to switch tables

### CODE REVIEW

File: cloudformation/membership-attribute-service.yaml [hunk 1]
```diff
@@ -71,8 +71,8 @@ Mappings:
     PROD:
       NotificationAlarmPeriod: 1200
       InstanceName: PROD:membership-attribute-service
-      DynamoDBTable: arn:aws:dynamodb:*:*:table/MembershipAttributes-PROD
-      DynamoDBTableTestUsers: arn:aws:dynamodb:*:*:table/MembershipAttributes-UAT
+      DynamoDBTable: "Memb-Attributes-Tables-PROD-SupporterAttributesFallback"
+      DynamoDBTableTestUsers: "Memb-Attributes-Tables-UAT-SupporterAttributesFallback"
```
Comment (pvighi) [inline cloudformation/membership-attribute-service.yaml]: these new table names assume that the tables cloudformation stacks were created with the name Memb-Attributes-Tables-[TOUCHPOINT_BACKEND]

File: cloudformation/membership-attribute-service-tables.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      TableName: !Sub SupporterAttributesFallback-${TouchpointBackend}
+      TimeToLiveSpecification:
+        AttributeName: "TTLTimestamp"
+        Enabled: true
+Outputs:
+  SupporterAttributesFallbackTable:
+    Description: supporter attributes fallback table
+    Value: !GetAtt [ SupporterAttributesFallbackTable, "Arn" ]
+    Export:
+      Name: !Sub ${AWS::StackName}-SupporterAttributesFallback
```
Comment (pvighi) [inline cloudformation/membership-attribute-service-tables.yaml]: the output parameter is exported prefixed with the name of the stack which seems to be a common practice and it's helpful for finding where an input parameter to another stack comes from. The only tradeoff that I see is that the name of the stack is manually set when it's created but it is written in the cloudformation templates that include this param so there is some room (but not a lot) for mistakes there.

I could also not use AWS::StackName in the  export name in which case we would have to search for the cfn template that includes the name to identify where an import comes from

File: cloudformation/membership-attribute-service-tables.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  TouchpointBackend:
+    Description: Environment name ( DEV, UAT or PROD)
+    Type: String
+    Default: DEV
+Mappings:
+  StageVariables:
+    DEV:
+          DynamoWriteCapacity: 1
+          DynamoReadCapacity: 5
+    UAT:
```
Comment (jacobwinch) [inline cloudformation/membership-attribute-service-tables.yaml]: Minor point, but the formatting is a bit wonky here

File: cloudformation/membership-attribute-service-tables.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          Fn::FindInMap: [ StageVariables, { Ref: TouchpointBackend }, DynamoReadCapacity ]
+      SSESpecification:
+        SSEEnabled: true
+      TableName: !Sub SupporterAttributesFallback-${TouchpointBackend}
+      TimeToLiveSpecification:
+        AttributeName: "TTLTimestamp"
+        Enabled: true
+Outputs:
+  SupporterAttributesFallbackTable:
+    Description: supporter attributes fallback table
```
Comment (jacobwinch) [inline cloudformation/membership-attribute-service-tables.yaml]: What about: "This Dynamo table is used to provide a failover if we cannot fetch a user's attributes from Zuora."

File: cloudformation/membership-attribute-service.yaml [hunk 5]
```diff
@@ -135,7 +136,8 @@ Resources:
             - dynamodb:BatchGetItem
             - dynamodb:DescribeTable
             Resource:
-              Fn::FindInMap: [ StageVariables, { Ref: Stage }, DynamoDBTableTestUsers ]
+              Fn::ImportValue:
```
Comment (jacobwinch) [inline cloudformation/membership-attribute-service.yaml]: On second thoughts, let's leave the permissions to read/write to the old tables alone. We can remove these when we are happy that we don't need to roll this change back (i.e. when we delete the old, unencrypted tables).

**Pull request merged**